src/main/java/ai/apptuit/metrics/jinsight/ConfigService.java

```diff
@@ -184,9 +188,44 @@ private static Properties loadProperties(File configFilePath) throws IOException
     try (InputStream inputStream = new BufferedInputStream(new FileInputStream(configFilePath))) {
       config.load(inputStream);
     }
+    // override with system properties, if present
+    config.putAll(loadSystemProperties());
     return config;
   }
 
+  private static Map<String, Object> loadSystemProperties() {
+    Map<String, Object> systemProperties = new HashMap<>();
+    String reporter = getProperty(REPORTER_SYSTEM_PROPERTY_NAME);
+    if(reporter != null){
+      systemProperties.put(REPORTER_PROPERTY_NAME, reporter);
+    }
+
+    String accessToken = getProperty(ACCESS_TOKEN_SYSTEM_PROPERTY_NAME);
+    if(accessToken != null){
+      systemProperties.put(ACCESS_TOKEN_PROPERTY_NAME, accessToken);
+    }
+
+    String apiEndpoint = getProperty(API_ENDPOINT_SYSTEM_PROPERTY_NAME);
+    if(apiEndpoint != null){
+      systemProperties.put(API_ENDPOINT_PROPERTY_NAME, apiEndpoint);
+    }
+
+    String globalTags = getProperty(GLOBAL_TAGS_SYSTEM_PROPERTY_NAME);
+    if(globalTags != null){
+      systemProperties.put(GLOBAL_TAGS_PROPERTY_NAME, globalTags);
+    }
+
+    return systemProperties;
+  }
+
+  private static String getProperty(String propertyName) {
+    String propertyValue = System.getenv(propertyName);
```

Comment:
if not propertyName.contains(".") then propertyName="jinsight."+propertyName

Then all new constants GLOBAL_TAGS_SYSTEM_PROPERTY_NAME REPORTER_SYSTEM_PROPERTY_NAME etc can be deleted